content/docs/rfcs/13/README.md

```diff
@@ -59,139 +59,260 @@ The following are not considered as part of the adversarial model:
+_"forward"_ indicates traversing the entries in ascending order,
+whereas _"backward"_ indicates traversing the entries in descending order.
+
+### Pagination
+
+If a large number of entries in the store service node match the query criteria provided in a `StoreRequest`,
+the client MAY make use of pagination
+in a chain of store request and response transactions
+to retrieve the full response in smaller batches termed _"pages"_.
+Pagination can be performed either in [a _forward_ or _backward_ direction](#waku-message-sorting).
+
+A store client MAY indicate the maximum number of matching entries it wants in the `StoreResponse`,
+by setting the page size limit in the `pagination_limit` field.
+Note that a store service node MAY enforce its own limit
+if the `pagination_limit` is unset
+or larger than the service node's internal page size limit.
+
+A `StoreResponse` with a populated `pagination_cursor` indicates that more stored entries match the query than included in the response.
+
+A `StoreResponse` without a populated `pagination_cursor` indicates that
+there are no more matching entries in the store.
+
+The client MAY request the next page of entries from the store service node
+by populating a subsequent `StoreRequest` with the `pagination_cursor` received in the `StoreResponse`. All other fields and query criteria MUST be the same as in the preceding `StoreRequest`.
+
+A `StoreRequest` without a populated `pagination_cursor` indicates that
+the client wants to retrieve the "first page" of the stored entries matching the query.
+
+## Store Request
+
+A client node MUST send all historical message queries within a `StoreRequest` message.
+This request MUST contain a `request_id`.
+The `request_id` MUST be a uniquely generated string.
+
+If the store client requires the store service node to include Waku message values in the query response,
+it MUST set `return_values` to `true`.
+If the store client requires the store service node to return only message hash keys in the query response,
+it SHOULD set `return_values` to `false`.
+By default, therefore, the store service node assumes `return_values` to be `false`.
+
+A store client MAY include query filter criteria in the `StoreRequest`.
+There are two types of filter use cases:
+1. Content filtered queries and
+2. Message hash lookup queries
+
+### Content filtered queries
+
+A store client MAY request the store service node to filter historical entries by a content filter.
+Such a client MAY create a filter on content topic, on time range or on both.
+
+To filter on content topic, the client MUST populate _both_ the `pubsub_topic` _and_ `content_topics` field.
+The client MUST NOT populate either `pubsub_topic` or `content_topics` and leave the other unset.
+Both fields MUST either be set or unset.
+A mixed content topic filter with just one of either `pubsub_topic` or `content_topics` set, SHOULD be regarded as an invalid request.
```

Comment:
Any specific reason for not allowing only `pubsubTopic` filter if we are allowing general StoreRequest without contentFilter itself?

Reply:
This is at least partly because within Waku, the `pubsubTopic` is simply a "routing qualifier" for specific content topics. In fact, every content filter scenario that we know about so far (also for Filter protocol) is designed to work on a set of a content topics. However, in order for these content topics to be "fully qualified" (i.e. globally unique) they need to be namespaced within a pubsub topic. All this is a long way of saying the Waku protocols are designed with this implicit assumption that pubsub topic does not (or should not) carry any "content information" so app features shouldn't be built in a way that violate this. For Filter this assumption also simplifies the filter matching quite a bit. Admittedly for Store it will be relatively simple to allow filtering on pubsub topic, but my assumption is that the exact same concept of "content filter" should be applicable (and interchangeable) between Filter and Store.

Of course, open to discuss loosening this requirement for Store if we are convinced about specific use case.

Reply:
I understand that for Filter it makes more sense as it is used only by nodes that use light-protocols. But Store queries are used by all types of nodes and was wondering if it helps in some scenarios to have pubsubTopic only filter. 

Few scenarios i can think of:
1. a Store node only serving a single pubsubTopic comes back online and wants to catch-up. In this case it can start with a history-query with that specific pubsubTopic and then do a sync to ensure it has all the messages. There are alternative ways to sync we can think of for this specific case, but this could be the simplest way.
2. A use-case with static sharding e.g like Status where-in they have too many content-topics to specify for a Store-Query and wouldn't it be easier if they use a filter with pubsubTopic only when the node comes online after being offline for some time. But, that may eventually end-up in fine-grain filtering at application/client once all messages are received.

Will think more on this and add if i see any additional scenarios for this specific query.